slip-0077.md

```diff
@@ -0,0 +1,62 @@
+# SLIP-0077 : Deterministic blinding key derivation for Confidential Transactions
+
+```
+Number:  SLIP-0077
+Title:   Deterministic blinding key derivation for Confidential Transactions
+Type:    Standard
+Status:  Draft
+Authors: Roman Zeyde <mail@romanzey.de>
+Created: 2019-06-15
+```
+
+## Abstract
+
+This document describes a method for blinding key derivation
+for Confidential Transactions, using a determinstic hierarchy.
+
+## General design
+
+In confidential transactions, the sender and the receiver use ECDH to derive a shared nonce, which is then used for hiding/recovering of the actual value and asset type being transacted.
+In Elements/Liquid, the receiver uses the following derivation scheme for his ECDH public/private keys:
+
+```
+blinding_private_key := HMAC_SHA256(key=master_blinding_key, msg=script_pubkey)
+blinding_public_key := secp256k1_publickey(private_key=blinding_private_key)
+```
+
+The receiver is using `blinding_public_key` construct a "blinded address", which is used by the sender to blind the relevant transaction outputs. Each such blinded transaction output also contains the sender's ECDH public key, so the receiver would be able to recover the shared nonce using its `blinding_private_key`.
+
+An additional use-case is sharing some/all of the receiver's blinding private keys with an external auditor, allowing unblinding the audited outputs without being able to spend them.
+
+## Design details
+
+### Master blinding key derivation
+
+In order to use similar blinding key derivation scheme on TREZOR, we suggest using the following [derivation scheme](https://docs.google.com/document/d/1QtwF8DF0TyPt9KFp5EsVpOY5d8PMicqafb2yGFoKfZ0/) for `master_blinding_key`:
+
+```
+domain := b"Symmetric key seed"
+root   := HMAC_SHA512(key=domain, msg=seed)
+
+label := b"SLIP-77"
+node  := HMAC_SHA512(key=root[0:32], msg=label)
```

Comment:
Assuming SLIP-0021 gets approved in the current form, add a null byte before the label, i.e. msg = b"\x00" + label.